ddtrace/propagation/http.py

```diff
@@ -246,6 +266,20 @@ def _extract(headers):
                 }
                 log.debug("failed to decode x-datadog-tags: %r", tags_value, exc_info=True)
 
+        if meta is not None and config._128_bit_trace_id_enabled:
+            # When 128 bit trace ids are propagated the 64 lowest order bits are encoded as an integer
+            # and set in the `x-datadog-trace-id` header (this was done for backwards compatibility).
+            # The 64 highest order bits are encoded in base 16 and store in the `_dd.p.tid` tag.
+            # Here we reconstruct the full 128 bit trace_id.
+            trace_id_hob_hex = meta.get(_HIGHER_ORDER_TRACE_ID_BITS)  # type: Optional[str]
+            if trace_id_hob_hex is not None:
+                # convert lowest order bits in trace_id to base 16
+                trace_id_lod_hex = "{:016x}".format(trace_id)
+                # combine highest and lowest order hex values to create a 128 bit trace_id
+                trace_id = int(trace_id_hob_hex + trace_id_lod_hex, 16)
+                # After the full trace id is reconstructed this tag is no longer required
```

Comment:
but we eventually set it again when we encode the trace?

Reply:
I made this change because I can see a scenario where the trace id in the context is manually overriden. Ie a user gets the current active context and does context.trace_id = 2**69. In this scenario we’ll keep a reference to a stale meta tag. Which isn’t a big deal but could lead to cryptic bugs if the code was refactored in the future.